File: app/src/main/java/de/amirrocker/testdagger2modules/base/injection/scopes/ActivityScope.java
package de.amirrocker.testdagger2modules.base.injection.scopes;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;
import javax.inject.Scope;

@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ActivityScope {
}

File: app/src/main/java/de/amirrocker/testdagger2modules/application/scope/ActivityScope.kt
package de.amirrocker.testdagger2modules.application.scope

import javax.inject.Scope


/**
 * We want to scope RegistrationViewModel to RegistrationComponent.
 * we have to annotate both the class and the interface with @ActivityScope.
 *
 */
@Scope
@MustBeDocumented
@Retention(value = AnnotationRetention.RUNTIME)
annotation class ActivityScope()
